simple-dns/src/dns/rdata/a.rs

```diff
@@ -19,6 +19,10 @@ impl<'a> PacketPart<'a> for A {
     where
         Self: Sized,
     {
+        if data.len() < *position + 4 {
+            return Err(crate::SimpleDnsError::InsufficientData);
+        }
```

Comment:
I am wondering if this can be a  bug elsewhere, because I believe the same check is done in the caller, [here](https://github.com/balliegojr/simple-dns/blob/f0b9c308a4a0bb14d60afb432310cf492a4d75b6/simple-dns/src/dns/rdata/macros.rs#L91)

A situation that may cause this is if the RData Len is < 4, which would be a bug in the conversion to wire format.

Do you still have the packet the caused the panic? I would like to take a look in the data if possible

Reply:
Unfortunately, I don't have packet that caused the issue. But I have stacktrace, maybe it will help:
```
2024-02-21T14:25:24.547651416+01:00 range end index 77 out of range for slice of length 73
2024-02-21T14:25:24.547659151+01:00 stack backtrace:
2024-02-21T14:25:24.561258199+01:00 0: rust_begin_unwind
2024-02-21T14:25:24.561299125+01:00 at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:645:5
2024-02-21T14:25:24.561307640+01:00 1: core::panicking::panic_fmt
2024-02-21T14:25:24.561314636+01:00 at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/panicking.rs:72:14
2024-02-21T14:25:24.561358338+01:00 2: core::slice::index::slice_end_index_len_fail_rt
2024-02-21T14:25:24.561365384+01:00 at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/slice/index.rs:76:5
2024-02-21T14:25:24.561372374+01:00 3: core::slice::index::slice_end_index_len_fail
2024-02-21T14:25:24.561378772+01:00 at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/slice/index.rs:68:9
2024-02-21T14:25:24.561386540+01:00 4: <simple_dns::dns::rdata::a::A as simple_dns::dns::packet_part::PacketPart>::parse
2024-02-21T14:25:24.561392889+01:00 5: <simple_dns::dns::rdata::RData as simple_dns::dns::packet_part::PacketPart>::parse
2024-02-21T14:25:24.561399685+01:00 6: <simple_dns::dns::resource_record::ResourceRecord as simple_dns::dns::packet_part::PacketPart>::parse
2024-02-21T14:25:24.561406099+01:00 7: simple_dns::dns::packet::Packet::parse_section
2024-02-21T14:25:24.561437278+01:00 8: simple_dns::dns::packet::Packet::parse
```
It happend on version 0.6.0

Reply:
@balliegojr I think I just encountered this issue. For updates, like below, the answer section could contain rdata with 0 `rdatalen`, and in which case, the check you mentioned above would have no effect, but the indexing in the rtype parser would fail.
In general, I think there is a lack of handling the 0 length rdata case, other than you had the negative indexing with `OPT`.

```
 ;; opcode: UPDATE, status: NOERROR, id: 42961
;; flags:; ZONE: 1, PREREQ: 1, UPDATE: 5, ADDITIONAL: 0

;; ZONE SECTION:
;blah.net.	IN	 SOA

;; PREREQUISITE SECTION:
test.blah.net.	0	NONE	CNAME	

;; UPDATE SECTION:
test.blah.net.	0	CLASS255	AAAA	
test.blah.net.	0	CLASS255	A	
test.blah.net.	1200	IN	AAAA	2607:f2c0:e4d9:f060::3fed
test.blah.net.	1200	IN	AAAA	2607:f2c0:e4d9:f060:dbb8:2ff1:55ac:da51
test.blah.net.	1200	IN	A	192.168.0.55

```

Reply:
@tugtugtug, yes, I think that may be the issue, this library really doesn't handle that scenario, and I was wondering if it was a faulty package or if the scenario was correct (having rdatalen == 0).

Thanks for pointing out the go library.

@Gr4b4rz sorry for taking this long to reply, time has been a limited resource lately. I will take a look at this next week, since I will be on vacation

Reply:
I have published a new version that handles empty records, which is probably the scenario that was causing the panic. @Gr4b4rz can you please check if it fixes the issue you had?

Reply:
Hey, thank you for quick reaction. I think this empty-rdata fix will solve my issue, so you can probably close this PR.